src/build.xml

```diff
@@ -152,19 +168,28 @@
         <requireVersion/> <!-- Must include a version property when task is executed. -->
         <echo file="${src.dir}/Spark/version.i" append="false" message="&amp;GLOBAL-DEFINE SPARK_VERSION ${release.version}-${release.date} (${oe.version})"/>
 
-        <!-- Code compiles without need for a database. -->
-        <PCTCompile destDir="${build.tmp}" preprocessDir="${preprocess}" dlcHome="${dlcHome}">
-            <propath>
-                <pathelement path="${src.dir}"/>
-                <pathelement path="${dlcHome}/tty/netlib/OpenEdge.Net.pl"/>
-                <pathelement path="${dlcHome}/tty/OpenEdge.Core.pl"/>
-            </propath>
-            <fileset dir="${src.dir}">
-                <include name="OpenEdge/**/*.cls"/>
-            </fileset>
-        </PCTCompile>
-
-        <!-- Code compiles without need for a database, but uses Ccs (from OE 11.7.4+) and any extra libs. -->
+        <if>
+            <equals arg1="${oeMajor}" arg2="11"/>
+            <then>
+                <!-- OpenEdge 11.7.x -->
+                <compile_openedge/>
+            </then>
+            <else>
+                <!-- OpenEdge 12.x -->
+                <if>
+                    <equals arg1="${oeMinor}" arg2="5"/>
+                    <then>
+                        <!-- OpenEdge 12.5 - Not Required -->
```

Comment:
And now I see why 12.5 is in the default .i ... why not just always compile? Why have special per-version cases?

Reply:
Crap, I'm trying to remember why this was necessary....

Reply:
This is because 11.7 and 12.2 need some updates to the BusinessLogic files which were released in 12.3 (or maybe later) which fixes some shortcomings of our CCS-BE implementation. By offering this as part of our demos we can get customers into a new pattern which should future-proof them for moving to 12.5 or whatever the next OE version may be.